addons/mrp/models/mrp_production.py

```diff
@@ -574,7 +574,8 @@ def _compute_workorder_ids(self):
             if production.state != 'draft':
                 continue
             workorders_list = [Command.link(wo.id) for wo in production.workorder_ids.filtered(lambda wo: not wo.operation_id)]
-            workorders_list += [Command.delete(wo.id) for wo in production.workorder_ids.filtered(lambda wo: wo.operation_id and wo.operation_id.bom_id != production.bom_id)]
+            deleted_workorders_ids = production.workorder_ids.filtered(lambda wo: wo.operation_id and wo.operation_id.bom_id != production.bom_id).ids
+            workorders_list += [Command.delete(wo_id) for wo_id in deleted_workorders_ids]
```

Comment:
I think it would be better to avoid deleting the line if it belongs to a sub BoM of the production order. That way, we won't delete/create the same work order and remove all the user data that could be edited.